ext/tk/tkutil/tkutil.c

```diff
@@ -167,14 +167,44 @@ tk_obj_untrust(VALUE self, VALUE obj)
   return obj;
 }
 
+struct tk_eval_cmd_call_args {
+    VALUE cmd;
+    int argc;
+    const VALUE *argv;
+};
+
+static VALUE
+tk_eval_cmd_call(VALUE arg)
+{
+    const struct tk_eval_cmd_call_args *args = (const struct tk_eval_cmd_call_args*)arg;
+    return rb_funcallv_kw(args->cmd, rb_intern("call"), args->argc, args->argv, 0);
+}
+
 static VALUE
 tk_eval_cmd(int argc, VALUE *argv, VALUE self)
 {
     VALUE cmd, rest;
 
     rb_scan_args(argc, argv, "1*", &cmd, &rest);
 #ifdef RB_PASS_KEYWORDS
+#  ifdef HAVE_RB_EVAL_CMD_KW
     return rb_eval_cmd_kw(cmd, rest, 0);
+#  else
+    if (RB_TYPE_P(cmd, T_STRING)) {
+        int pstate;
+        return rb_eval_string_protect(RSTRING_PTR(cmd), &pstate);
+    } else {
+        int pstate;
+        struct tk_eval_cmd_call_args args;
+        VALUE val;
+        args.cmd = cmd;
+        args.argc = RARRAY_LENINT(rest);
+        args.argv = RARRAY_CONST_PTR(rest);
+        val = rb_protect(tk_eval_cmd_call, (VALUE)&args, &pstate);
+        RB_GC_GUARD(rest);
+        return val;
+    }
```

Comment:
`rb_eval_cmd_kw()` in https://github.com/ruby/ruby/blob/941e70ab38d01d067b7bbbcdf8553893a9ca8b49/vm_eval.c#L2148-L2179 doesn't suppress exceptions. The `rb_protect()` around the eval and funcall should be unnecessary.

Reply:
Thank you. I misunderstood the purpose of the tag code in `rb_eval_cmd_call_kw` when trying to recreate the behavior. I'll update the PR.